python/cudf/cudf/core/series.py

```diff
@@ -2187,43 +2187,32 @@ def quantile(
 
         """
 
-        if isinstance(q, Number) or is_list_like(q):
-            np_array_q = np.asarray(q)
-            if np.logical_or(np_array_q < 0, np_array_q > 1).any():
-                raise ValueError(
-                    "percentiles should all \
-                             be in the interval [0, 1]"
-                )
-
-        # Beyond this point, q either being scalar or list-like
-        # will only have values in range [0, 1]
+        q_np = np.asarray(q)
+        if (q_np < 0).any() or (q_np > 1).any():
+            raise ValueError(
+                "percentiles should all be in the interval [0, 1]"
+            )
 
-        if isinstance(q, Number):
-            res = self._column.quantile(q, interpolation, exact)
-            if len(res) == 0:
+        if len(self) == 0:
+            if isinstance(q, Number):
                 return np.nan
-            else:
-                # if q is an int/float, we shouldn't be constructing series
-                return res.pop()
 
-        if not quant_index:
-            return Series(
-                self._column.quantile(q, interpolation, exact), name=self.name
-            )
-        else:
-            from cudf.core.column import column_empty_like
-
-            np_array_q = np.asarray(q)
-            if len(self) == 0:
-                result = column_empty_like(
-                    np_array_q,
-                    dtype=self.dtype,
-                    masked=True,
-                    newsize=len(np_array_q),
+            if quant_index:
+                return build_column(
+                    dtype=(self.dtype if exact else np.dtype("float64"))
                 )
-            else:
-                result = self._column.quantile(q, interpolation, exact)
-            return Series(result, index=as_index(np_array_q), name=self.name)
+
+        result = self.__class__._from_table(
+            self._column.quantile(q, interpolation, not exact)
+        )
```

Comment:
I would expect this to return a `Column` and for us to construct from a `Column` here

Reply:
I'm confused. `Series.quantile` should return a `Column`?

Reply:
I would expect `_column.quantile(...)` to return a Column as opposed to a `Frame`

Reply:
Or remove `column.quantile` altogether as you pointed out above and just use `self.quantile` to hit the `Frame` implementation.